packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -30,3 +31,38 @@ export function logIfFalse(
 	logger.send(newEvent);
 	return false;
 }
+
+/**
+ * sampletext
+ */
+export function createSampledLoggerSend(
+	logger: ITelemetryBaseLogger,
+	shouldSampleEventCallback: () => boolean,
+) {
+	const mc = createChildMonitoringContext({ logger });
+	const isSamplingEnabled = mc.config.getBoolean("Fluid.Telemetry.DisableSampling");
+
+	return (event: ITelemetryBaseEvent) => {
+		if (isSamplingEnabled && shouldSampleEventCallback() === true) {
+			logger.send(event);
+		}
+	};
+}
+
+/**
+ * sampletext
+ */
+export const createSystematicSamplingCallback = (samplingRate: number) => {
+	const state = {
+		eventsSinceLastSample: 0,
+		samplingRate,
```

Comment:
Seems unnecessary here if we're creating a closure over the `samplingRate` parameter in the returned function, which seems ok. Using `state.samplingRate` below would also be fine, but not sure it makes much of a difference.

Reply:
Agreed. Updated this.